### PR TITLE
Support rpm packages

### PR DESCRIPTION
**Pull request by zimathias**
# Issue
Refactor cpack configuration so that both rpm and debian packages can be built.
## Tasklist
 - [ ] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [x] review
 - [x] adjust for comments

### CODE REVIEW

**Comment by TheMarex** [comment]
@zimathias looks good to me. Could you rebase on master through? I see there is a merge commit in this branch. Thanks. 👍

File: cmake/CPackConfig.cmake [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+string(TOLOWER "${CMAKE_PROJECT_NAME}" CPACK_PACKAGE_NAME)
+SET(CPACK_PACKAGE_VERSION_MAJOR ${OSRM_VERSION_MAJOR})
+SET(CPACK_PACKAGE_VERSION_MINOR ${OSRM_VERSION_MINOR})
+SET(CPACK_PACKAGE_VERSION_PATCH ${OSRM_VERSION_PATCH})
+SET(CPACK_PACKAGE_VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH}")
+
+SET(CPACK_INCLUDE_TOPLEVEL_DIRECTORY "FALSE")
+SET(CPACK_PACKAGE_DESCRIPTION_FILE "${CMAKE_SOURCE_DIR}/README.md")
+SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Open Source Routing Machine (OSRM) is a high-performance routing engine. It combines sophisticated routing algorithms with the open and free data of the OpenStreetMap.")
+SET(CPACK_PACKAGE_CONTACT "Dennis Luxen <info@project-osrm.org>")
```
Comment (daniel-j-h) [inline cmake/CPackConfig.cmake]: I think we should change that. It's two years now that Dennis is no longer involved with OSRM development.

File: CMakeLists.txt [hunk 2]
```diff
@@ -697,8 +698,8 @@ install(TARGETS osrm_store DESTINATION lib)
 # Setup exporting variables for pkgconfig and subproject
 #
 
-if(BUILD_DEBIAN_PACKAGE)
-  include(CPackDebianConfig)
+if(BUILD_PACKAGE)
```
Comment (oxidase) [inline CMakeLists.txt]: Would it make sense to add `option(BUILD_PACKAGE "Build OSRM package" OFF)`  on line 20?

**Comment by daniel-j-h** [comment]
@zimathias could you rebase on master and adapt your changes to the remarks left as comments.

File: CMakeLists.txt (hunk 2 above)
Comment (zimathias) [inline CMakeLists.txt]: done.

**Comment by zimathias** [comment]
@daniel-j-h @TheMarex I've addressed the comments and rebased the branch.

**Pull request merged**